### ISSUE TITLE
Tech 1 weapons show incorrect attributes when switching from tech 2 weapon and charge using variations menu

### ISSUE THREAD

**Issue by Candyguitar** [pyfa-org/Pyfa#2270]
## Bug Report
If a tech 2 weapon is loaded with tech 2 charges, switching to a tech 1 variation causes it to show the base attributes instead of modified. This is very noticeable on attack battlecruisers with their role bonuses for large weapons.

### Expected behavior:
Using the variations menu, switching to a tech 1 weapon from a tech 2 version should always show the attributes modified by character skills and ship bonuses.

### Actual behavior:
If one of the tech 2 weapons had tech 2 charges, the last tech 1 weapon will use base attributes instead of modified. This results in incorrect calculations for CPU, power grid, range, etc. The rest (if applicable) are calculated correctly. Closing and reopening the fit won't fix it, but changing it in any way will even after it is undone; for example, fitting another module and then pressing CTRL-Z.

### Detailed steps to reproduce:

1. Start with any ship fitted with at least one tech 2/faction/storyline/officer weapon and tech 2 charge.
2. Right click on the weapon or list of weapons and select a tech 1 variation.

### Fits involved in EFT format (Edit > To Clipboard > EFT):

```
[Oracle, Weapon attribute bug test]

[Empty Low slot]
[Empty Low slot]
[Empty Low slot]
[Empty Low slot]
[Empty Low slot]
[Empty Low slot]

[Empty Med slot]
[Empty Med slot]
[Empty Med slot]

Polarized Mega Pulse Laser, Conflagration L
Polarized Mega Pulse Laser, Conflagration L
[Empty High slot]
[Empty High slot]
[Empty High slot]
[Empty High slot]
[Empty High slot]
[Empty High slot]

[Empty Rig slot]
[Empty Rig slot]
[Empty Rig slot]

```
```
[Tornado, Weapon attribute bug test]

[Empty Low slot]
[Empty Low slot]
[Empty Low slot]
[Empty Low slot]

[Empty Med slot]
[Empty Med slot]
[Empty Med slot]
[Empty Med slot]
[Empty Med slot]

1200mm Artillery Cannon II, Quake L
1200mm Artillery Cannon II, Quake L
[Empty High slot]
[Empty High slot]
[Empty High slot]
[Empty High slot]
[Empty High slot]
[Empty High slot]

[Empty Rig slot]
[Empty Rig slot]
[Empty Rig slot]
```
```
[Kestrel, Weapon attribute bug test]

[Empty Low slot]
[Empty Low slot]

[Empty Med slot]
[Empty Med slot]
[Empty Med slot]
[Empty Med slot]

Rocket Launcher II, Mjolnir Rage Rocket
Rocket Launcher II, Mjolnir Rage Rocket
[Empty High slot]
[Empty High slot]

[Empty Rig slot]
[Empty Rig slot]
[Empty Rig slot]
```
```
[Enyo, Weapon attribute bug test]

[Empty Low slot]
[Empty Low slot]
[Empty Low slot]
[Empty Low slot]

[Empty Med slot]
[Empty Med slot]
[Empty Med slot]

150mm Railgun II, Spike S
150mm Railgun II, Spike S
[Empty High slot]
[Empty High slot]
[Empty High slot]

[Empty Rig slot]
[Empty Rig slot]
```
### Release or development git branch? Please note the release version or commit hash:
Version 2.28.2.

### Operating system and version (eg: Windows 10, OS X 10.9, OS X 10.11, Ubuntu 16.10):
Windows 10 and Linux Mint 19.

### Other relevant information:
I fixed the issue and will submit a pull request for it. While I still haven't worked out why only one weapon is affected, the fix works around it by removing tech 2 charges from tech 1 weapons in eos.saveddata.module.calculateModifiedAttributes.

**Comment by DarkFenX** [comment]
Thanks for the report. I've started noticing the same issue as well but did not look close enough into it, to look for what caused it. I believe it was one of optimizations which removed "unnecessary" recalcs.

Your fix seems to be not the right way to handle it. So i will close the PR and look into proper way to fix it. Thanks for looking into it, though.

**Comment by DarkFenX** [comment]
Fixed in e27ea79bc6333a36fc79aececcc3010f222959cb
